tests/core/v5_1/test_v51_rpc_handlers.py

```diff
@@ -73,8 +100,99 @@ def ping_param_w3(request, alice, bob, bob_network):
 
 
 @pytest.mark.trio
-async def test_rpc_ping_web3(make_request, ping_param_w3, alice, bob, w3):
-    pong = await trio.to_thread.run_sync(w3.discv5.ping, ping_param_w3)
+async def test_v51_rpc_ping(make_request, bob_node_id_param, alice, bob):
+    pong = await make_request("discv5_ping", [bob_node_id_param])
+    assert pong["enr_seq"] == bob.enr.sequence_number
+    assert pong["packet_ip"] == inet_ntoa(alice.endpoint.ip_address)
+    assert pong["packet_port"] == alice.endpoint.port
+
+
+@pytest.mark.trio
+async def test_v51_rpc_ping_invalid_node_id(make_request, invalid_node_id, alice, bob):
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_ping", [invalid_node_id])
+    with pytest.raises(Exception, match="'error':"):
+        await make_request("discv5_ping", [])
+
+
+@pytest.mark.trio
+async def test_v51_rpc_ping_web3(make_request, bob_node_id_param_w3, alice, bob, w3):
+    pong = await trio.to_thread.run_sync(w3.discv5.ping, bob_node_id_param_w3)
     assert pong.enr_seq == bob.enr.sequence_number
     assert pong.packet_ip == ipaddress.ip_address(alice.endpoint.ip_address)
     assert pong.packet_port == alice.endpoint.port
+
+
+@pytest.mark.trio
+async def test_v51_rpc_findNodes(make_request, bob_node_id_param, alice, bob):
+    distances = set()
+
+    for _ in range(10):
+        enr = ENRFactory()
+        distances.add(compute_log_distance(bob.node_id, enr.node_id))
+        bob.enr_db.set_enr(enr)
+
+    # request with positional single distance
+    enrs_at_0 = await make_request("discv5_findNodes", [bob_node_id_param, 0])
+    assert all(isinstance(ENR.from_repr(enr), ENR) for enr in enrs_at_0)
```

Comment:
Same thing on these lines, I think the `isinstance` is extraneous, and I think bucket 0 will always be empty. I think this test could pick an element for `distances`.